test/e2e/config/azure-dev.yaml

```diff
@@ -169,6 +180,7 @@ variables:
   AKS_KUBERNETES_VERSION_UPGRADE_FROM: "latest-1"
   KUBERNETES_VERSION: "${KUBERNETES_VERSION:-stable-1.27}"
   FLATCAR_KUBERNETES_VERSION: "${FLATCAR_KUBERNETES_VERSION:-stable-1.26}"
+  UPGRADE_KUBERNETES_VERSION: "v1.26.12"
```

Comment:
would it be better to call this `UPGRADE_FROM_KUBERNETES_VERSION`?

Reply:
we already have `KUBERNETES_VERSION_UPGRADE_FROM` so we should make a distinction if there is one, otherwise it gets confusing

Reply:
Renamed it to `UPGRADE_TEST_KUBERNETES_VERSION` for clarity

Reply:
The other one is also for an "upgrade test" no? I think we need to clarify API Upgrade test vs workload upgrade test